lib/component.js

```diff
@@ -68,11 +69,29 @@ function build ( string, opts, readFile ) {
 						promises.push( readFile( src ).then( str => partials[id] = str ) );
 					}
 				} else if ( !type || type === 'text/javascript' || type === 'application/javascript' ) {
+					const rel = getAttr( 'rel', item);
+
+					if ( rel === 'css' ) {
+						if ( cssFn ) return Promise.reject( `Only one css script may be specified per component.` );
+						else if ( style.length ) return Promise.reject( `<style> elements and css script cannot be mixed.` );
```

Comment:
They could probably mix. They could concat into one big string in the order they appear, and the end result is the function version so that the functions would still work.

```
<style>
  .foo{ color: red }
</style>
<style type="text/ractive-css-fn">
  .bar{ color: ${ this.barColor }
</style>

// Becomes
css: function(){
  return `
    .foo{ color: red }
    .bar{ color: ${ this.barColor }
  `
}
```

But it shouldn't work if a `type` is present and contains something other than the function CSS type. That way, the tool can warn if say Sass was present.

```
<style type="text/scss">
  .foo{ color: $fooColor }
</style>
<style type="text/ractive-css-fn">
  .bar{ color: ${ this.barColor }
</style>
```

My 2c anyways. Alternatively, I've always been a fan of a single-`<style>` like the single-`<script>`. I find the spec strange that it allows for multiple `<style>` in a component file.